File: TestTaskBrio/ClientApp/src/app/data/models/Point.ts
import { ICircle } from "./ICircle";

export class Marker implements ICircle {

    x: number;
    y: number;
    radius: number;
    nextMarker: any;

    constructor(x: number, y: number, radius: number) {
        this.x = x;
        this.y = y;
        this.radius = radius;
    }
}

export class NeighborMarker {
    x: number;
    y: number;

    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }
}

File: TestTaskBrio/Point.cs
namespace TestTaskBrio
{
    public class Marker
    {
        public float X { get; set; }
        public float Y { get; set; }
        public int Radius { get; set; }
        public NeighborMarker PreviousMarker { get; set; }
        public NeighborMarker NextMarker { get; set; }
    }

    public class NeighborMarker
    {
        public float X { get; set; }
        public float Y { get; set; }
    }
}
